### ISSUE TITLE
assigning i18n properties to Ember Objects.

### ISSUE THREAD

**Issue by alexanderjeurissen** [jamesarosen/ember-i18n#329]
I currently have an issue trying to assign i18n translated text to properties in a `Ember.object`

I can access the `TranslationMacro` inside the `Ember.Object` which is enough for my usecase, however when trying to access the property from another Ember object I get the following error:

<img width="1279" alt="screen shot 2015-11-03 at 16 25 20" src="https://cloud.githubusercontent.com/assets/1220084/10911973/a7a18fee-8247-11e5-8a3d-76a7656db807.png">

This is the code of the Ember object:
```javascript
var UploadStrategy = Ember.Object.extend({
  i18n: Ember.inject.service(),

  accept: 'image/*',
  allowedType: 'image/',
  allowedSize: Constants.MAX_HEADER_FILE_SIZE,
  method: 'POST',
  multiple: false,
  invalidFileSizeMessage: t('errors.invalidImageSize', {
      allowedSize: (Constants.MAX_HEADER_FILE_SIZE / 1000000) + 'MB'
  }),
  invalidFileTypeMessage: t('errors.invalidImageType'),
  url: function (file) {
    return 'attachments';
  }
});

export default UploadStrategy;
```

The idea is that I can create an instance of the UploadStrategy and when necessary overwrite the messages.

for example in this file I create an instance of the UploadStrategy and want to retrieve the `invalidFileSizeMessage` in order to reject a promise with a localised error message.

```javascript
import Ember from 'ember';
import EmberUploader from 'ember-uploader';
import UploadStrategy from 'learning-spaces/lib/upload-strategy';
import Constants from 'learning-spaces/constants';

var Uploader = EmberUploader.Uploader.extend(Ember.Evented, {
  validationStrategy: UploadStrategy.create(),

  upload: function (file, data) {

    if (!this.hasValidFileSize(file, this.validationStrategy.allowedSize)) {
      var error = new Error(this.validationStrategy.get('invalidFileSizeMessage'));
      return new Promise(function (resolve, reject) {
        reject({ errorThrown: error });
      }.bind(this));
    }

    return this._super(file, data);
  }
});

export default Uploader;
```

**Comment by jamesarosen** [comment]
How are you creating instances of the object? If you're not asking the Container for them, you'll need to pass in a `container` manually.

**Comment by alexanderjeurissen** [comment]
you mean how I create instances of the `UploadStrategy` ? it's the line `validationStrategy: UploadStrategy.create()`

If your question is how the Uploader get's instantiated then I'll have to check some additional code.

**Comment by jamesarosen** [comment]
That's what I meant. You'll need to pass it in a container. Unfortunately, you don't have one in the context you're currently creating it. You'll have to defer creation until inside `upload`.

**Comment by alexanderjeurissen** [comment]
hmm ok, I'll give that a try tomorrow. Thanks for your help so far.

**Comment by jamesarosen** [comment]
You could do
```js
validationStrategy: Ember.computed(function() {
  return UploadStrategy.create({
    container: this.container
  })
})
```

**Comment by jamesarosen** [comment]
Did you get this working?

**Comment by alexanderjeurissen** [comment]
I haven't had the time to test your solution as of yet. When I do I'll let you know.

**Comment by alexanderjeurissen** [comment]
@jamesarosen I didn't get the i18n to work due to the complexity of the upload strategies and how they were used elsewhere in the application. 

the Upload strategy was being extended by other ember objects that replaced the error messages for specific upload scenarios.

In the end I decided to not translate the values inside the upload strategy but to just store the key, and the params as an object in the upload strategy and then translate it inside the upload method of the uploader.

your tip about the containers did help me though, as I had to use it wherever the Uploader was instantiated. 

lesson learned: If it's really hard to get some code working then there is probably a better way to implement it. The tight coupling of the strategies is a sign that it needs to be refactored in the near future but my solution will do for now.

Thanks for your help
